CMakeLists.txt

```diff
@@ -1276,7 +1280,7 @@ target_compile_options(unicorn-common PRIVATE
     ${UNICORN_COMPILE_OPTIONS}
 )
 # necessary to prevent the redefinition of the symbol crc32 on some compilers
-target_compile_definitions(unicorn-common PRIVATE crc32=uc_crc32 gen_helper_crc32=gen_helper_uc_crc32)
+target_compile_definitions(unicorn-common PRIVATE "$<$<NOT:$<C_COMPILER_ID:GNU>>:crc32=uc_crc32>" gen_helper_crc32=gen_helper_uc_crc32)
```

Comment:
Shouldn't gen_helper_crc32 be redefined only if crc32 is redefined?
Also how is this causing issues? I can understand it not being needed on MinGW but I don't see how adding it would cause compilation issues.

Reply:
1. I updated the code.

2. It leads to this error:
```
[build] C:\WINDOWS\system32\cmd.exe /C "cd . && C:\msys64\mingw64\bin\c++.exe -DTRACY_NO_CRASH_HANDLER -g  @CMakeFiles\vita3k.rsp -o bin\Vita3K.exe -Wl,--out-implib,lib\libVita3K.dll.a -Wl,--major-image-version,0,--minor-image-version,0 && C:\WINDOWS\system32\cmd.exe /C "cd /D E:\build-vita3k-gcc-debug\vita3k && C:\msys64\mingw64\bin\cmake.exe -E copy_directory C:/src/Vita3K/vita3k/../data E:/build-vita3k-gcc-debug/bin/data && C:\msys64\mingw64\bin\cmake.exe -E copy_directory C:/src/Vita3K/vita3k/../lang E:/build-vita3k-gcc-debug/bin/lang && C:\msys64\mingw64\bin\cmake.exe -E copy_directory C:/src/Vita3K/vita3k/shaders-builtin E:/build-vita3k-gcc-debug/bin/shaders-builtin && C:\msys64\mingw64\bin\cmake.exe -E copy_if_different C:/src/Vita3K/external/sdl/windows/lib/x64/SDL2.dll E:/build-vita3k-gcc-debug/bin && cd /D E:\build-vita3k-gcc-debug\vita3k && C:\msys64\mingw64\bin\cmake.exe -E copy_if_different C:/src/Vita3K/vita3k/script/update-windows.bat E:/build-vita3k-gcc-debug/bin/update-vita3k.bat""
[build] C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: external/unicorn/libarm-softmmu.a(helper.c.obj): in function `helper_crc32_arm':
[build] C:/src/Vita3K/external/unicorn/qemu/target/arm/helper.c:11450:(.text+0x15836): undefined reference to `crc32'
[build] collect2.exe: error: ld returned 1 exit status
```

Probably because for GCC the order of libraries on the command line matters and `libz` is in the wrong place.

Reply:
Could you put it inside a if(NOT MINGW) instead of using cmake generator expressions please?

Reply:
It is not MINGW specific, it is specific to GCC, but I can use `IF(NOT CMAKE_C_COMPILER_ID STREQUAL "GNU")`.

Reply:
done